src/hooks/useGet.js

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+import { useQuery } from 'react-query';
+
+function refreshNoop() {
+  console.warning(
+    'refresh is not implemented for useGet - replace with react-query invalidation logic instead.',
+  );
+}
+
+// to do: borrow some logic from error formatter in utils, then delete that function
+function formatError(response) {
+  try {
+    return response?.error ? response.error.toJSON().message : null;
+  } catch (e) {
+    return 'Error could not be formatted as JSON';
+  }
+}
+
+export default function useGet({
+  queryKey,
+  url,
+  data,
+  dataAccessor = result => result?.data?.data,
+  onSuccess = Function.prototype,
+  prependHoustonApiUrl = true,
+  queryOptions = {},
+}) {
+  const [displayedError, setDisplayedError] = useState(null);
+  const [statusCode, setStatusCode] = useState(null);
+
+  const apiUrl = prependHoustonApiUrl
+    ? `${__houston_url__}/api/v1${url}`
+    : url;
+  const result = useQuery(
+    queryKey,
+    async () => {
+      const response = await axios.request({
+        url: apiUrl,
+        method: 'get',
+        data,
+      });
+      const status = response?.status;
+      setStatusCode(status);
+      if (status === 200) onSuccess(response);
+      return response;
+    },
+    {
+      staleTime: Infinity,
```

Comment:
Yeah, that's what I was thinking. Usually in our app data expires because of an action, not because a certain amount of time has passed. The only exception I could think of was `GET ags/sighting`, so I figure better to use this as the dfault and override in that case.

Reply:
How would one override this??

Reply:
something like `queryOptions: { staleTime: 5000 }` would do the trick. Because queryOptions are spread onto the end so they override the defaults

Reply:
Very cool. Good to know. Thank you.